spyder/api/plugin_registration/registry.py

```diff
@@ -317,6 +344,147 @@ def notify_plugin_availability(self, plugin_name: str,
                 plugin_instance = self.plugin_registry[plugin]
                 plugin_instance._on_plugin_available(plugin_name)
 
+    def delete_plugin(self, plugin_name: str) -> bool:
+        """
+        Remove and delete a plugin from the registry by its name.
+
+        Paremeters
+        ----------
+        plugin_name: str
+            Name of the plugin to delete.
+
+        Returns
+        -------
+        plugin_deleted: bool
+            True if the registry was able to teardown and remove the plugin.
+            False otherwise.
+        """
+        plugin_instance = self.plugin_registry[plugin_name]
+
+        # Determine if plugin can be closed
+        can_delete = True
+        if isinstance(plugin_instance, SpyderPluginV2):
+            can_delete = plugin_instance.can_close()
+        elif isinstance(plugin_instance, SpyderPlugin):
+            can_delete = plugin_instance.closing_plugin(True)
+
+        if not can_delete:
+            return False
+
+        if isinstance(plugin_instance, SpyderPluginV2):
+            # Disconnect plugin from other plugins
+            self._teardown_plugin(plugin_name)
+
+            # Disconnect depending plugins from the plugin to delete
+            self._notify_plugin_teardown(plugin_name)
+
+            # Remove the plugin from the main window (if graphical)
+            if isinstance(plugin_instance, SpyderDockablePlugin):
+                plugin_instance.close_window()
+
+            # Perform plugin closure tasks
+            plugin_instance.on_close(True)
+        elif isinstance(plugin_instance, SpyderPlugin):
+            # Disconnect depending plugins from the plugin to delete
+            self._notify_plugin_teardown(plugin_name)
+            if isinstance(plugin_instance, SpyderPluginWidget):
+                plugin_instance._close_window()
+
+        # Delete plugin from the registry and auxiliary structures
+        self.plugin_dependents.pop(plugin_name, None)
+        self.plugin_dependencies.pop(plugin_name, None)
+
+        for plugin_ in self.plugin_dependents:
```

Comment:
I find the naming of `plugin_` a bit odd. Why not call that variable simply `plugin`?